Gems/ScriptCanvas/Code/Include/ScriptCanvas/Execution/RuntimeComponent.cpp

```diff
@@ -18,167 +17,58 @@
-    }
-
-    GraphIdentifier RuntimeComponent::GetGraphIdentifier() const
-    {
-        return GraphIdentifier(AZ::Data::AssetId(m_runtimeOverrides.m_runtimeAsset.GetId().m_guid, 0), 0);
-    }
-
-    AZ::EntityId RuntimeComponent::GetScriptCanvasId() const
-    {
-        return m_scriptCanvasId;
-    }
-
-    const RuntimeDataOverrides& RuntimeComponent::GetRuntimeDataOverrides() const
-    {
-        return m_runtimeOverrides;
-    }
-
-    void RuntimeComponent::TakeRuntimeDataOverrides(RuntimeDataOverrides&& overrideData)
-    {
-        m_runtimeOverrides = AZStd::move(overrideData);
-        m_runtimeOverrides.EnforcePreloadBehavior();
-    }
-
-    void RuntimeComponent::Init()
-    {
-        m_scriptCanvasId = AZ::Entity::MakeId();
-        AZ_Assert(RuntimeDataOverrides::IsPreloadBehaviorEnforced(m_runtimeOverrides), "RuntimeComponent::m_runtimeAsset Auto load behavior MUST be set to AZ::Data::AssetLoadBehavior::PreLoad");
-    }
-
-    void RuntimeComponent::InitializeExecution()
-    {
-#if defined(SCRIPT_CANVAS_RUNTIME_ASSET_CHECK)
-        if (!m_runtimeOverrides.m_runtimeAsset.Get())
-        {
-            AZ_Error("ScriptCanvas", false, "RuntimeComponent::m_runtimeAsset AssetId: %s was valid, but the data was not pre-loaded, so this script will not run"
-                , m_runtimeOverrides.m_runtimeAsset.GetId().ToString<AZStd::string>().data());
-            return;
-        }
-#else
-        AZ_Assert(m_runtimeOverrides.m_runtimeAsset.Get(), "RuntimeComponent::m_runtimeAsset AssetId: %s was valid, but the data was not pre-loaded, so this script will not run"
-            , m_runtimeOverrides.m_runtimeAsset.GetId().ToString<AZStd::string>().data());
-#endif
-
-        AZ_PROFILE_SCOPE(ScriptCanvas, "RuntimeComponent::InitializeExecution (%s)", m_runtimeOverrides.m_runtimeAsset.GetId().ToString<AZStd::string>().c_str());
-        SCRIPT_CANVAS_PERFORMANCE_SCOPE_INITIALIZATION(m_scriptCanvasId, m_runtimeOverrides.m_runtimeAsset.GetId());
-        m_executionState = ExecutionState::Create(ExecutionStateConfig(m_runtimeOverrides.m_runtimeAsset, *this));
-
-#if defined(SCRIPT_CANVAS_RUNTIME_ASSET_CHECK)
-        if (!m_executionState)
-        {
-            AZ_Error("ScriptCanvas", false, "RuntimeComponent::m_runtimeAsset AssetId: %s failed to create an execution state, possibly due to missing dependent asset, script will not run"
-                , m_runtimeOverrides.m_runtimeAsset.GetId().ToString<AZStd::string>().data());
-            return;
-        }
-#else
-        AZ_Assert(m_executionState, "RuntimeComponent::m_runtimeAsset AssetId: %s failed to create an execution state, possibly due to missing dependent asset, script will not run"
-            , m_runtimeOverrides.m_runtimeAsset.GetId().ToString<AZStd::string>().data());
-#endif
-
-        AZ::EntityBus::Handler::BusConnect(GetEntityId());
-        m_executionState->Initialize();
+        const auto entityId = GetEntityId();
```

Comment:
All of the RuntimeComponent implementation that doesn't have to do directly with the Entity/Component interface has been moved to Executor.